### ISSUE TITLE
Safeguard for identity map seems to break merge

### ISSUE THREAD

**Issue by alexander-schranz** [doctrine/doctrine2#10820]
I'm currently running into the following issues since: https://github.com/doctrine/orm/pull/10785

<img width="1180" alt="Bildschirmfoto 2023-07-06 um 16 18 50" src="https://github.com/doctrine/orm/assets/1698337/36f230cb-f860-4a8b-a14a-c5023a9f8dc3">

### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | ^2@dev (current released version works)

#### Summary

For me it looks this currently breaks the existing `EntityManager::merge` method. I know merge is deprecated but it is used in this case as the object get not retriefied from the database instead unserialized and then crashes when it tries to `merge` into the entityManager. https://github.com/sulu/sulu/blob/f977ffc0f480cb1d4ddee9cde0cbb8a881011d91/src/Sulu/Bundle/AudienceTargetingBundle/Entity/TargetGroupRepository.php#L33

Should not merged entities not show this identiy map warning as they got merged and only persisted?

#### Previous behavior

Don't did crash on current implemented method.

#### Current behavior

Crashes with a new method.

#### How to reproduce

```
git clone git@github.com:sulu/sulu.git
composer update doctrine/orm:"@dev" --prefer-source

# start database
# export DATABASE_URL=...

bin/runtests -i -t AudienceTargetingBundle -C --flags="--filter=testPut"
```

**Comment by mpdude** [comment]
Can you provide an isolated test case for ORM? 

To avoid regressions, we need to have that test in our own suite, we cannot run Sulu's tests in our CI.

**Comment by mpdude** [comment]
The object you’re trying to merge (which has been serialized previously, if I get you right), is that a proxy instance?

**Comment by alexander-schranz** [comment]
I tested 2.17.3 and 2.17.4 today. It seems like the issue does not longer exist also I tested https://github.com/doctrine/orm/issues/10867 which seems also be fixed there I know it was fixed in 2.16.1.
